### PR TITLE
Handle paths with a question mark in them

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This needs https://github.com/quarkusio/quarkus-fs-util/pull/14 to be merged and 0.0.9 to be released.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/classloader/ClassLoadingPathTreeResourceUrlTestCase.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     *
+     * @throws Exception
+     * @see <a href="https://github.com/quarkusio/quarkus/issues/11707"/>
+     */
+    @Test
+    public void testResourceAsStreamForDirectory() throws Exception {
+        final JavaArchive jar = ShrinkWrap.create(JavaArchive.class)
+                .add(new StringAsset("a"), "a.txt")
+                .add(new StringAsset("b"), "b/b.txt");
+        final Path tmpDir = Files.createTempDirectory("test?");
```
Comment (gastaldi) [inline independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/classloader/ClassLoadingPathTreeResourceUrlTestCase.java:66]: Isn't `?` illegal in Windows?

Comment (gastaldi) [reply independent-projects/bootstrap/core/src/test/java/io/quarkus/bootstrap/classloader/ClassLoadingPathTreeResourceUrlTestCase.java:66]: Yes, just confirmed that this test will fail in Windows

**Comment by aloubyansky** [comment]
I restarted the CI

**Review by aloubyansky — APPROVED** [review]

**Comment by aloubyansky** [comment]
Failure on Windows
````
2022-02-15T10:33:41.8008350Z [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.1 s <<< FAILURE! - in io.quarkus.bootstrap.classloader.ClassLoadingResourceUrlTestCase
2022-02-15T10:33:41.8281692Z [ERROR] io.quarkus.bootstrap.classloader.ClassLoadingResourceUrlTestCase.testUrlReturnedFromClassLoaderDirectory(String)[1]  Time elapsed: 0.02 s  <<< ERROR!
2022-02-15T10:33:41.8283828Z java.nio.file.InvalidPathException: Illegal char <?> at index 4: test?15515974475056451777
2022-02-15T10:33:41.8284596Z 	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
2022-02-15T10:33:41.8285236Z 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
2022-02-15T10:33:41.8287601Z 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
2022-02-15T10:33:41.8288209Z 	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
2022-02-15T10:33:41.8288768Z 	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
2022-02-15T10:33:41.8289612Z 	at java.base/java.nio.file.TempFileHelper.generatePath(TempFileHelper.java:59)
2022-02-15T10:33:41.8290205Z 	at java.base/java.nio.file.TempFileHelper.create(TempFileHelper.java:126)
2022-02-15T10:33:41.8290826Z 	at java.base/java.nio.file.TempFileHelper.createTempDirectory(TempFileHelper.java:172)
2022-02-15T10:33:41.8291433Z 	at java.base/java.nio.file.Files.createTempDirectory(Files.java:1007)
2022-02-15T10:33:41.8292391Z 	at io.quarkus.bootstrap.classloader.ClassLoadingResourceUrlTestCase.testUrlReturnedFromClassLoaderDirectory(ClassLoadingResourceUrlTestCase.java:39)
````

**Comment by famod** [comment]
This needs a rebase to receive the latest flaky test fixes.

**Pull request merged**